### ISSUE TITLE
Slideshow notices in debug logfile

### ISSUE THREAD

**Issue by nerdykit** [netPhotoGraphics/netPhotoGraphics#1093]
I'm seeing some of this in my debug logfile after upgrading to 2.00.03
Not sure if it's a problem in the slideshow code or if it's a problem at my end.
Thanks

NK

 NOTICE: "Undefined offset: 117" in /mnt/usbdrive/webroot/public_html/gallery/npgCore/extensions/slideshow.php on line 228

URI:/gallery/page/slideshow

IP `99.226.202.233`

theme:basic_for_kit

slideshow::getShow called from printSlideShow (slideshow.php [773])

from include (slideshow.php [36])

from include (index.php [164])

from index.php [68]

{19225:Wed, 13 Nov 2019 05:50:53 GMT}	

NOTICE: "Undefined offset: 117" in /mnt/usbdrive/webroot/public_html/gallery/npgCore/extensions/slideshow.php on line 233

URI:/gallery/page/slideshow

IP `99.226.202.233`

theme:basic_for_kit

slideshow::getShow called from printSlideShow (slideshow.php [773])

from include (slideshow.php [36])

from include (index.php [164])

from index.php [68]

{19225:Wed, 13 Nov 2019 05:50:53 GMT}	

Bad filename suffix in newImage()

URI:/gallery/page/slideshow

IP `99.226.202.233`

theme:basic_for_kit

newImage called from slideshow::getShow (slideshow.php [234])

from printSlideShow (slideshow.php [773])

from include (slideshow.php [36])

from include (index.php [164])

from index.php [68]

{19225:Wed, 13 Nov 2019 05:50:53 GMT}	

NOTICE: "Undefined offset: 118" in /mnt/usbdrive/webroot/public_html/gallery/npgCore/extensions/slideshow.php on line 228

URI:/gallery/page/slideshow

IP `99.226.202.233`

theme:basic_for_kit

slideshow::getShow called from printSlideShow (slideshow.php [773])

from include (slideshow.php [36])

from include (index.php [164])

from index.php [68]

{19225:Wed, 13 Nov 2019 05:50:53 GMT}	

NOTICE: "Undefined offset: 118" in /mnt/usbdrive/webroot/public_html/gallery/npgCore/extensions/slideshow.php on line 233

URI:/gallery/page/slideshow

IP `99.226.202.233`

theme:basic_for_kit

slideshow::getShow called from printSlideShow (slideshow.php [773])

from include (slideshow.php [36])

from include (index.php [164])

from index.php [68]

{19225:Wed, 13 Nov 2019 05:50:53 GMT}	

Bad filename suffix in newImage()

URI:/gallery/page/slideshow

IP `99.226.202.233`

theme:basic_for_kit

newImage called from slideshow::getShow (slideshow.php [234])

from printSlideShow (slideshow.php [773])

from include (slideshow.php [36])

from include (index.php [164])

from index.php [68]

**Comment by sbillard** [comment]
This is a rather strange error report. You can ignore the "bad filename suffix" error as it is a result of the undefined offset. But that error is not something I can easily explain.

The code being executed is
`if (is_array($images[$idx])) {
				$filename = $images[$idx]['filename'];
				$album = newAlbum($images[$idx]['folder']);
				$image = newImage($album, $filename);
			} else {
				$filename = $images[$idx];
				$image = newImage($albumobj, $filename);
			}`

with the line `$filename = $images[$idx];` throwing the error. BUT, the IF statement has fetched `$images[$idx]` apparently without error. 

So, some questions. 

Have you seen the error when viewing your site? (If not maybe the next few questions are unanswerable.)

The "117" is presumably the index of the image within the slideshow that is being accessed. Do you know how many images there should have been? 

If you do not have the shuffle option set ($shuffle parameter in the printSlideShow() call, defaults to false/NULL) then image will be 118th in the album.

Do any images show in the slideshow? Does the slideshow work on other albums?

**Comment by nerdykit** [comment]
The IP address is not my own so the site was being browsed by someone else.  I've looked at my directories of photos and I can find only one directory that holds that many photos (124 actually) with about five being movies.  So I ran the slideshow for that directory and saw no visible problems and the debug log shows no new entries.  The only thing I did notice was that the "count" that appears in the title line  3/124  4/124 etc. occasionally seemed to get confused (I think at one point it stepped backward about 10 places) but the slideshow was otherwise fine.  Since I can't actually see any problem here, I think I'll not waste any more of your time on it.  I thought that perhaps I'd noticed a bug in the slideshow area of nPG but perhaps none exists.  I'll continue to monitor debug log and re-open this if something else comes up there.

Thanks

NK
